### ISSUE TITLE
Different search output w/o ucinewgame

### ISSUE THREAD

**Issue by joergoster** [official-stockfish/Stockfish#1101]
With the current master, I get different search output depending whether a 'ucinewgame' command is sent or not.

Example:
```
uciok
go depth 6
info depth 1 seldepth 1 multipv 1 score cp 91 nodes 20 nps 5000 tbhits 0 time 4 pv e2e4
info depth 2 seldepth 2 multipv 1 score cp 93 nodes 47 nps 11750 tbhits 0 time 4 pv e2e4 b7b6
info depth 3 seldepth 3 multipv 1 score cp 118 nodes 124 nps 31000 tbhits 0 time 4 pv d2d4 e7e6 e2e4
info depth 4 seldepth 4 multipv 1 score cp 108 nodes 253 nps 63250 tbhits 0 time 4 pv d2d4 h7h6 e2e4 e7e6
info depth 5 seldepth 5 multipv 1 score cp 31 nodes 1460 nps 292000 tbhits 0 time 5 pv e2e4 b8c6 f1c4 c6e5 c4d5
info depth 6 seldepth 6 multipv 1 score cp 4 nodes 3988 nps 569714 tbhits 0 time 7 pv g1f3 e7e6 e2e3 d7d5 d2d4 b8c6
bestmove g1f3 ponder e7e6
```
```
uciok
ucinewgame
isready
readyok
go depth 6
info depth 1 seldepth 1 multipv 1 score cp 91 nodes 20 nps 10000 tbhits 0 time 2 pv e2e4
info depth 2 seldepth 2 multipv 1 score cp 93 nodes 47 nps 23500 tbhits 0 time 2 pv e2e4 b7b6
info depth 3 seldepth 3 multipv 1 score cp 118 nodes 124 nps 62000 tbhits 0 time 2 pv d2d4 e7e6 e2e4
info depth 4 seldepth 4 multipv 1 score cp 108 nodes 251 nps 125500 tbhits 0 time 2 pv d2d4 h7h6 e2e4 e7e6
info depth 5 seldepth 5 multipv 1 score cp 41 nodes 1656 nps 552000 tbhits 0 time 3 pv b1c3 e7e6 g1f3 d7d5 d2d4
info depth 6 seldepth 6 multipv 1 score cp 4 nodes 3296 nps 824000 tbhits 0 time 4 pv b1c3 d7d5 d2d4 c8g4 g1f3 e7e6
bestmove b1c3 ponder d7d5
```
*
Older versions don't suffer from this behaviour.
(e.g. version from March 20, 2017)
```
uciok
go depth 6
info depth 1 seldepth 1 multipv 1 score cp 88 nodes 20 nps 5000 tbhits 0 time 4 pv e2e4
info depth 2 seldepth 2 multipv 1 score cp 90 nodes 47 nps 11750 tbhits 0 time 4 pv e2e4 b7b6
info depth 3 seldepth 3 multipv 1 score cp 121 nodes 128 nps 32000 tbhits 0 time 4 pv d2d4 d7d6 e2e4
info depth 4 seldepth 4 multipv 1 score cp 36 nodes 547 nps 109400 tbhits 0 time 5 pv d2d4 e7e6 e2e4 d7d5
info depth 5 seldepth 5 multipv 1 score cp 48 nodes 966 nps 193200 tbhits 0 time 5 pv d2d4 e7e6 b1d2 d7d5 e2e4
info depth 6 seldepth 6 multipv 1 score cp 0 nodes 3052 nps 436000 tbhits 0 time 7 pv g1f3 d7d5 d2d4 g8f6 c1g5 b8c6
bestmove g1f3 ponder d7d5
```
```
uciok
ucinewgame
isready
readyok
go depth 6
info depth 1 seldepth 1 multipv 1 score cp 88 nodes 20 nps 10000 tbhits 0 time 2 pv e2e4
info depth 2 seldepth 2 multipv 1 score cp 90 nodes 47 nps 23500 tbhits 0 time 2 pv e2e4 b7b6
info depth 3 seldepth 3 multipv 1 score cp 121 nodes 128 nps 64000 tbhits 0 time 2 pv d2d4 d7d6 e2e4
info depth 4 seldepth 4 multipv 1 score cp 36 nodes 547 nps 273500 tbhits 0 time 2 pv d2d4 e7e6 e2e4 d7d5
info depth 5 seldepth 5 multipv 1 score cp 48 nodes 966 nps 322000 tbhits 0 time 3 pv d2d4 e7e6 b1d2 d7d5 e2e4
info depth 6 seldepth 6 multipv 1 score cp 0 nodes 3052 nps 763000 tbhits 0 time 4 pv g1f3 d7d5 d2d4 g8f6 c1g5 b8c6
bestmove g1f3 ponder d7d5
```
*
The reason for this is most likely this https://github.com/official-stockfish/Stockfish/commit/ced29248c93de7fc5a4e284807f8f052006e647c patch.
Table is filled by Search::clear() after 'ucinewgame', but not at engine startup.

From the UCI specs:

>    If the GUI hasn't sent a "ucinewgame" before the first "position" command, the engine shouldn't
>    expect any further ucinewgame commands as the GUI is probably not supporting the ucinewgame command.
>    So the engine should not rely on this command even though all new GUIs should support it.

**Comment by vondele** [comment]
yes, will have a look.... I noticed this when you were working on the other issue, and I vaguely recall what the issue is.

**Comment by vondele** [comment]
should be fixed with #1104

**Comment by joergoster** [comment]
Thanks!
